l10n_jp_partner_title_qweb/models/res_partner.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    # we make title a computed field instead of using onchange to update it
+    # because onchange is not triggered when a user signs up from portal
+    title = fields.Many2one(
+        compute='_compute_title',
+        store=True,
+        readonly=False,
+    )
+
+    @api.multi
+    @api.depends('lang', 'is_company')
+    def _compute_title(self):
+        for partner in self:
+            if not partner.lang or self._is_linked_to_company(
+                    self.commercial_company_name):
+                partner.title = False
+            else:
+                lang_id = self.env['res.lang'].search(
+                    [('code', '=', partner.lang)], limit=1).id
+                title = self.env['res.partner.title'].search(
+                    [('lang_id', '=', lang_id),
+                     ('for_company', '=', partner.is_company)], limit=1)
+                partner.title = title.id if title else False
```

Comment:
Sorry, my explanation was insufficient.
I mean that the code above should be as follows I think...

```python
if title:
    partner.title = title.id
```

Reply:
@eternalharvest The result will actually be the same since a compute method starts with clearing assigned values of the concerned fields. :)